doc/playbook.rst

```diff
@@ -1,108 +1,267 @@
-
-Available Row Generators
-------------------------
-
-EvenRowGenerator
-^^^^^^^^^^^^^^^^
-
-Generates a cluster of even numbered rows.  Uses the row mapping passed down from the PayloadGenerator.  Here are the configs that can be specified in *row_generator_config*\ :
-
-*nr_rows*
-
-Number of rows to generate
+Payload generator for measuring and characterizing rowhammer tolerance.
+It can provide information about how many rows and bits are susceptible to rowhammer.
+It can also provide information about where the susceptible bits are located.
+
+A series of double-sided hammers against the available group of victim rows is performed.
+The double-sided hammers increase in intensity based on ``read_count_step`` parameter.
+Here are the parameters that can be specified in *payload_generator_config* for this payload generator:
+
+* ``row_mapping`` - this is the :ref:`row mapping` used
+* ``row_generator`` - this is the :ref:`row generator class` used to generate the rows
+* ``row_generator_config`` - parameters for the row generator
+* ``verbose`` - should verbose output be generated (true or false)
+* ``fill_local`` - when enabled, permits shrinking the filled memory area to just the aggressors and the victim
+* ``nr_rows`` - number of rows to conduct the experiment over. This is the number of aggressor rows.
+  Victim rows will be 2 times fewer than this number. For example, to perform hammering for 32 victim rows, use 34 as the parameter value
+* ``read_count_step`` - this is how much to increment the hammer count between multiple tests for the same row.
+  This is the number of hammers on single side (total number of hammers on both sides is 2x this value)
+* ``iters_per_row`` - number of times the hammer count is incremented for each row
+
+The results are a series of histograms with appropriate labeling.
+
+Example :ref:`configurations` for this test were provided as ``configs/example_hammer_*.cfg`` files.
+Some of them require a significant amount o memory declared as `payload memory`.
+To execute a minimalistic example from within litex-rowhammer-tester repo enter:
+
+.. code-block:: console
+
+  source venv/bin/activate
+  cd rowhammer_tester/scripts/playbook/
+  python playbook.py configs/example_hammer_minimal.cfg
+
+Expected output:
+
+.. code-block:: console
+
+  Progress: [========================================] 3072 / 3072 
+  Generating payload:
+    tRAS = 5
+    tRP = 3
+    tREFI = 782
+    tRFC = 32
+    Repeatable unit: 186
+    Repetitions: 93
+    Payload size =  0.04KB /  1.00KB
+    Payload per-row toggle count = 0.010K  x2 rows
+    Payload refreshes (if enabled) = 10 (disabled)
+    Expected execution time = 1263 cycles = 0.013 ms
+  
+  Transferring the payload ...
+  
+  Executing ...
+  Time taken: 0.647 ms
+  
+  Progress: [============                            ]  323 / 1024 (Errors: 320) 
+  ...
+
+Configurations
+--------------
+
+Test configuration files are represented as python dictionaries. An example:
```

Comment:
We may just say they are JSON files as we now use the `json` module to load them. Maybe we should also change file extensions to `.json` to avoid confusion.

Reply:
Applied these suggestions